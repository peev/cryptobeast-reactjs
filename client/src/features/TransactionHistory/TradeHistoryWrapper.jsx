// @flow
import React from 'react';
import { withStyles } from 'material-ui';

import { RegularCard } from '../../components';
import TradeHistory from './TradeHistory/TradeHistory';
// import ExportPdfButton from '../../components/CustomButtons/ExportPdfButton';


const styles = () => ({
  tablePosition: {
    marginTop: '80px',
  },
});

type Props = {
  classes: Object,
};


const TradeHistoryWrapper = ({ classes }: Props) => (
  <RegularCard
    cardTitle="TRADE HISTORY"
    // button={<ExportPdfButton />}
    content={
      <React.Fragment>
        <div className={classes.tablePosition}>
          <TradeHistory
            tableHead={[
              'Trade Date',
              'Entry date',
              'Source',
              'Pair',
              'Type',
              'Price',
              'Filled',
              'Fee',
              'Total',
              '',
              '',
            ]}
          />
        </div>
      </React.Fragment>
    }
  />
);


export default withStyles(styles)(TradeHistoryWrapper);
