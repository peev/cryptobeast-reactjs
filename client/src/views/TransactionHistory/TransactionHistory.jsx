// @flow
import React from 'react';
import { withStyles, Grid } from 'material-ui';

import { ItemGrid } from './../../components';
import TradeHistory from '../../components/Cards/History/TradeHistory';
import NotificationSnackbar from '../../components/Modal/NotificationSnackbar';


const style = {
  typo: {
    paddingLeft: '25%',
    marginBottom: '40px',
    position: 'relative',
  },
  note: {
    fontFamily: '"Lato", "Helvetica", "Arial", sans-serif',
    bottom: '10px',
    color: '#c0c1c2',
    display: 'block',
    fontWeight: '400',
    fontSize: '13px',
    lineHeight: '13px',
    left: '0',
    marginLeft: '20px',
    position: 'absolute',
    width: '260px',
  },
};
const TransactionHistory = ({ classes }: Object) =>
  (<Grid container>
      <ItemGrid xs={12} sm={12} md={12}>
        <p>FILTERS</p>
      </ItemGrid>
      <Grid container className={classes.itemsCardPosition}>
      <Grid item xs={12} sm={12} md={12}>
        <TradeHistory />
      </Grid>
      </Grid>
      <NotificationSnackbar />
   </Grid>);


export default withStyles(style)(TransactionHistory);
