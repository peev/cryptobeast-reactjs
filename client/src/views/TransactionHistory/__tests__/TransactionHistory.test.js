import React from 'react';
import { shallow } from 'enzyme';
import TransactionHistory from './../TransactionHistory';

describe('Views/TransactionHistory.jsx', () => {
  const transactionHistory = shallow(<TransactionHistory />);

  it('should render without crashing', () => {
    expect(transactionHistory).toMatchSnapshot();
  });
});
