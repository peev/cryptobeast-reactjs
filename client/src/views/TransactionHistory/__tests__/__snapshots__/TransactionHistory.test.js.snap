// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Views/TransactionHistory.jsx should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(TransactionHistory) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "note": "TransactionHistory-note-02",
        "typo": "TransactionHistory-typo-01",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "note": "TransactionHistory-note-02",
          "typo": "TransactionHistory-typo-01",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
